api/src/main/java/org/motechproject/nms/api/web/KilkariController.java

```diff
@@ -101,4 +104,73 @@ public void saveInboxCallDetails(@RequestBody InboxCallDetailsRequest request) {
         //todo: the work...
     }
 
+    /**
+     * 4.2.3
+     * Create Subscription
+     *
+     * IVR shall invoke this API to create a new Kilkari subscription
+     *
+     */
+    @RequestMapping(value = "/subscription",
+            method = RequestMethod.POST,
+            headers = { "Content-type=application/json" })
+    @ResponseStatus(HttpStatus.OK)
+    public void createSubscription(@RequestBody SubscriptionRequest subscriptionRequest)
+        throws NotFoundException {
+
+        StringBuilder failureReasons = validate(subscriptionRequest.getCallingNumber(),
+                subscriptionRequest.getOperator(), subscriptionRequest.getCircle(),
+                subscriptionRequest.getCallId());
+        validateFieldPresent(failureReasons, "subscriptionPack", subscriptionRequest.getSubscriptionPack());
+        validateFieldPresent(failureReasons, "languageLocationCode",
+                subscriptionRequest.getLanguageLocationCode().toString());
+
+        if (failureReasons.length() > 0) {
+            throw new IllegalArgumentException(failureReasons.toString());
+        }
+        if (!validateSubscriptionPack(subscriptionRequest.getSubscriptionPack())) {
+            throw new NotFoundException(String.format(NOT_FOUND, "subscriptionPack"));
+        }
+
+        kilkariService.createSubscription(subscriptionRequest.getCallingNumber(),
+                subscriptionRequest.getLanguageLocationCode(), subscriptionRequest.getSubscriptionPack());
+    }
+
+    /**
+     * 4.2.4
+     * Deactivate Subscription
+     *
+     * IVR shall invoke this API to deactivate an existing Kilkari subscription
+     *
+     */
+    @RequestMapping(value = "/subscription",
+            method = RequestMethod.DELETE,
```

Comment:
Following the spec :)

-----Original Message-----
From: "Rob LaRubbio" <notifications@github.com>
Sent: ‎4/‎22/‎2015 4:42 PM
To: "koshalt/mim" <mim@noreply.github.com>
Cc: "Lauren Lavoie" <llavoie@grameenfoundation.org>
Subject: Re: [mim] Kilkari CreateSubscription and DeactivateSubscription (#73)

In api/src/main/java/org/motechproject/nms/api/web/KilkariController.java:
> +            throw new NotFoundException(String.format(NOT_FOUND, "subscriptionPack"));
> +        }
> +
> +        kilkariService.createSubscription(subscriptionRequest.getCallingNumber(),
> +                subscriptionRequest.getLanguageLocationCode(), subscriptionRequest.getSubscriptionPack());
> +    }
> +
> +    /**
> +     * 4.2.4
> +     * Deactivate Subscription
> +     *
> +     * IVR shall invoke this API to deactivate an existing Kilkari subscription
> +     *
> +     */
> +    @RequestMapping(value = "/subscription",
> +            method = RequestMethod.DELETE,
Should you really have a delete method with a post body?  :)
—
Reply to this email directly or view it on GitHub.